lib/trivial.nix

```diff
@@ -448,6 +448,39 @@ rec {
   isFunction = f: builtins.isFunction f ||
     (f ? __functor && isFunction (f.__functor f));
 
+  /* Copy the function arguments (__functionArgs) of the first function
+     onto the second function.
+
+     Type:
+       copyFunctionArgs :: ({ <a_i> } -> b) -> (a -> c) -> (({ <a_i> } -> c) & AttrSet)
+
+     Example:
+       addab = {a, b}: a + b
+       addab { a = 2; b = 4; }
+       => 6
+       lib.functionArgs addab
+       => { a = false; b = false; }
+       addab1 = attrs: addab attrs + 1
+       addab1 { a = 2; b = 4; }
+       => 7
+       lib.functionArgs addab1
+       => { }
+       addab1' = lib.copyFunctionArgs addab addab1
+       addab1' { a = 2; b = 4; }
+       => 7
+       lib.functionArgs addab1'
+       => { a = false; b = false; }
+   */
+  copyFunctionArgs =
+    # Function to provide the argument metadata
+    f:
+    let
+      fArgs = functionArgs f;
+    in
+    # Function to set the argument metadata to
```

Comment:
Side note: This doesn't render in nixdoc right now, but I think we can keep it anyways

Reply:
I noticed that, too. It would be fine as long as the function description doesn't depend on the argument name used here.

Nevertheless, I do hope that the second one could get rendered in the future.